numpy/core/src/multiarray/mapping.c

```diff
@@ -2616,7 +2617,8 @@ PyArray_MapIterNew(npy_index_info *indices , int index_num, int index_type,
     PyArrayObject *original_extra_op = extra_op;
 
     PyArrayObject *index_arrays[NPY_MAXDIMS];
-    PyArray_Descr *dtypes[NPY_MAXDIMS];
+    PyArray_Descr *intp_descr = PyArray_DescrFromType(NPY_INTP);
```

Comment:
Missing null check?

Reply:
It actually cannot fail for these builtin types, since it is just a table lookup + incref. Of course, maybe that is an implementation detail (it probably was used sloppy around that area more often, but not sure).

Reply:
Perhaps just add an `assert` then

Reply:
I am adding the error check, but I am wondering if it may make sense to give the no-error guarantee (for our own numeric dtypes).